File: 1380-lucky-numbers-in-a-matrix/1380-lucky-numbers-in-a-matrix.py
class Solution:
    def luckyNumbers (self, matrix: List[List[int]]) -> List[int]:
        answer = []
        count, min_idx = 0, 0
        
        for i, m in enumerate(matrix):
            min_idx, count = 0, 0
            for j in range(1, len(m)):
                if m[j] < m[min_idx]:
                    min_idx = j
            # print("행의 최솟값:", m[min_idx])
            
            for m2 in matrix:
                # print(m2[min_idx], "값과 비교", end="\t")
                if m2[min_idx] > m[min_idx]:
                    break
                count += 1
            
            if count == len(matrix):
                answer.append(m[min_idx])
        return answer

File: 1380-lucky-numbers-in-a-matrix/1380-lucky-numbers-in-a-matrix.cpp
class Solution {
public:
    vector<int> luckyNumbers (vector<vector<int>>& matrix) {
        vector<int> answer;
        int minIdx, count;
        for(int i = 0; i < matrix.size(); i++) { //행 순회
            minIdx = 0;
            count = 0;
            
            for(int j = 1; j < matrix[i].size(); j++) { //행의 최솟값 찾기
                if(matrix[i][j] < matrix[i][minIdx]) minIdx = j;
            }
            
            for(int j = 0; j < matrix.size(); j++) { //열의 최댓값 찾기
                if(matrix[j][minIdx] > matrix[i][minIdx]) break;
                count++;
            }
            
            if(count == matrix.size()) {
                answer.push_back(matrix[i][minIdx]);
            }
        }
        
        return answer;
    }
};